src/test/java/hudson/plugins/active_directory/docker/ActiveDirectoryGenericContainer.java

```diff
@@ -0,0 +1,29 @@
+package hudson.plugins.active_directory.docker;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.InternetProtocol;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+public class ActiveDirectoryGenericContainer<SELF extends ActiveDirectoryGenericContainer<SELF>> extends GenericContainer<SELF> {
+
+    public ActiveDirectoryGenericContainer() {
+        setImage(new ImageFromDockerfile("ad-dc", false)
+                .withFileFromClasspath("custom.sh", "hudson/plugins/active_directory/docker/TheFlintstonesTest/TheFlintstones/custom.sh")
+                .withFileFromClasspath("Dockerfile", "hudson/plugins/active_directory/docker/TheFlintstonesTest/TheFlintstones/Dockerfile")
+                .withFileFromClasspath("init.sh", "hudson/plugins/active_directory/docker/TheFlintstonesTest/TheFlintstones/init.sh")
+                .withFileFromClasspath("kdb5_util_create.expect", "hudson/plugins/active_directory/docker/TheFlintstonesTest/TheFlintstones/kdb5_util_create.expect")
+                .withFileFromClasspath("krb5.conf", "hudson/plugins/active_directory/docker/TheFlintstonesTest/TheFlintstones/krb5.conf")
+                .withFileFromClasspath("named.conf.options", "hudson/plugins/active_directory/docker/TheFlintstonesTest/TheFlintstones/named.conf.options")
+                .withFileFromClasspath("sssd.conf", "hudson/plugins/active_directory/docker/TheFlintstonesTest/TheFlintstones/sssd.conf")
+                .withFileFromClasspath("supervisord.conf", "hudson/plugins/active_directory/docker/TheFlintstonesTest/TheFlintstones/supervisord.conf"));
+        setWaitStrategy(null);
+        addFixedExposedPort(135, 135);
+        addFixedExposedPort(138, 138);
+        addFixedExposedPort(445, 445);
+        addFixedExposedPort(39, 39);
+        addFixedExposedPort(464, 464);
+        addFixedExposedPort(389, 389);
+        addFixedExposedPort(3268, 3268);
+        addFixedExposedPort(53, 53, InternetProtocol.UDP);
```

Comment:
Kindly refrain from commenting about pre-existing issues on my PRs.

Reply:
Anyone reviewing this would also see you are using fixed ports in this OR which is a red flag and would have to redo the investigation.

Sure my review maybe went a little deeper but without a comment from you saying `qas already fixed ports see loc` I will continue to leave these comments for others as an aid to review.

Reply:
The old code used fixed ports, and so does the new code. I have not changed the status quo. If you do not like the status quo, then do something about it (such as filing a PR), but please do not comment about your dislike of the status quo on my PR when I did not bring about the current status quo.

Reply:
Basil you have misread my PR and my comment please reread them.

Reply:
My reading comprehension skills are not in need of improvement.

Reply:
🤷

Reply:
Do you intend to file a PR or at least an issue to track the future work you have in mind?

Reply:
it is in progress.  However I am also hitting several issues in TestContainers that make them less functional that our own docker fixtures.